component/espejote-templates/rules-renderer-v1.libsonnet

```
@@ -0,0 +1,217 @@
+
+local procPatchExprUserWorkload(group, config) =
+  local dropUserWorkload(alert) = std.member(config.ignoreUserWorkload, alert);
+  local patchThingy(field) = std.foldl(
+    function(a, b) b(a),
+    [
+      function(e)
+        std.strReplace(
+          e,
+          'namespace=~"openshift-monitoring|openshift-user-workload-monitoring"',
+          'namespace="openshift-monitoring"'
+        ),
+      function(e)
+        std.strReplace(
+          e,
+          'job=~"alertmanager-main|alertmanager-user-workload"',
+          'job="alertmanager-main"'
+        ),
+      function(e)
+        std.strReplace(
+          e,
+          'job=~"prometheus-k8s|prometheus-user-workload"',
+          'job="prometheus-k8s"'
+        ),
+    ],
+    field
+  );
+  group {
+    rules: [
+      if dropUserWorkload(std.get(rule, 'alert', '')) then
+        rule {
+          expr: patchThingy(super.expr),
+          // I have no clue what this part is originally for.
+          // I know what it does, just not why...
+          [if std.objectHas(std.get(rule, 'annotations', {}), 'description') then 'annotations']+: {
+            description: patchThingy(super.description),
+          },
+        }
+      else
+        rule
+      for rule in super.rules
+    ],
+  };
+
+local procPatchExprSelector(group, config) =
+  local markerOpenshiftNamespace = 'namespace=~"(openshift-.*|kube-.*|default)"';
+  // Replace namespace selector in expression
+  local hasNamespaceSelector = std.length(config.renderedIncludeSelector) > 0;
+  local hasExclusionSelector = std.length(config.renderedExcludeSelector) > 0;
+  group {
+    rules: [
+      rule {
+        expr: std.strReplace(
+          super.expr,
+          markerOpenshiftNamespace,
+          (
+            if hasNamespaceSelector then
+              'namespace=~"(%s)"' % config.renderedIncludeSelector
+            else ''
+
+          ) + (
+            if hasNamespaceSelector && hasExclusionSelector then ',' else ''
+          ) + (
+            if hasExclusionSelector then
+              'namespace!~"(%s)"' % config.renderedExcludeSelector
+            else ''
+          )
+        ),
+      }
+      for rule in super.rules
+    ],
+  };
+
+local procFilterRules(group, config) =
+  local getSeverity(r) = std.get(std.get(r, 'labels', {}), 'severity', '');
+  local drop(r) =
+    // drop rules with severity info
+    (
+      getSeverity(r) == 'info'
+    ) ||
+    // drop rules with severity warning that are in the ignoreWarnings list
+    (
+      getSeverity(r) == 'warning' &&
+      std.member(
+        config.ignoreWarnings,
+        std.get(r, 'alert', '')
+      )
+    ) ||
+    // drop rules with alert name in the ignoreNames list
+    (
+      std.member(
+        config.ignoreNames,
+        std.get(r, 'alert', '')
+      )
+    );
+  group {
+    rules: std.filter(function(rule) !drop(rule), super.rules),
+  };
+
+local processGroup(group, config) = std.foldl(
+  function(a, b) b(a, config),
```

might be clearer with more descriptive function args:

```suggestion
  function(g, func) func(g, config),
```